src/ProjectTemplates/Web.ProjectTemplates/content/BlazorWeb-CSharp/BlazorWeb-CSharp/Components/Identity/LogoutForm.razor

```diff
@@ -0,0 +1,34 @@
+﻿@using Microsoft.AspNetCore.Identity
+@using BlazorWeb_CSharp.Data
+@using BlazorWeb_CSharp.Identity
+
+@inject SignInManager<ApplicationUser> SignInManager
+@inject NavigationManager NavigationManager
+@inject IdentityRedirectManager RedirectManager
+
+<form @onsubmit="OnSubmitAsync" @formname="logout" method="post" @attributes="AdditionalAttributes">
+    <AntiforgeryToken />
+    <input type="hidden" name="ReturnUrl" value="@NavigationManager.Uri" />
+</form>
+
+@code {
+    [Parameter(CaptureUnmatchedValues = true)]
+    public IDictionary<string, object?>? AdditionalAttributes { get; set; }
```

Comment:
Why are we passing through additional attributes here? The developer can just add their own attributes directly on line 9 since this is in their own source code.

I see you're using it to pass `id`, but why is that needed?

Reply:
The ID is needed so we can submit the form via JavaScript. I acknowledge it's not the most intuitive thing so we could definitely consider another approach. See https://github.com/dotnet/aspnetcore/pull/50722#discussion_r1334595887